pyiceberg/table/puffin.py

```diff
@@ -0,0 +1,118 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import math
+from typing import TYPE_CHECKING, Dict, List, Literal, Optional
+
+from pydantic import Field
+from pyroaring import BitMap, FrozenBitMap
+
+from pyiceberg.typedef import IcebergBaseModel
+
+if TYPE_CHECKING:
+    import pyarrow as pa
+
+# Short for: Puffin Fratercula arctica, version 1
+MAGIC_BYTES = b"PFA1"
+EMPTY_BITMAP = FrozenBitMap()
+MAX_JAVA_SIGNED = int(math.pow(2, 31))
```

Comment:
```suggestion
MAX_JAVA_SIGNED = int(math.pow(2, 31)) - 1
```

https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html#MAX_VALUE